### DISCUSSION TITLE
How to plot the IPF image in such a way that specific {hkl} planes are within 10° to either RD, ND or TD?

### DISCUSSION THREAD

**Discussion by kopparthiravikiran** [mtex-toolbox/mtex#1902] [Ask Anything]
I would like to plot the IPF image in such a way that specific {hkl} planes are within 10° of any plane. Please find the below image.

![image007](https://github.com/mtex-toolbox/mtex/assets/128254603/73c49e99-7943-4401-b2a2-750d6c4fb122)

In HKL Channel 5, this can be done, find the below screenshot.

![image009](https://github.com/mtex-toolbox/mtex/assets/128254603/10025ff0-a439-4bae-9050-752fa517cfcb)

I am unable to plot this in MTEX.

Thank you very much,
Ravi.

**Comment by kilir** [comment]
Hi,
I'm not entirely sure if I understand what you want to do, but maybe you want to plot points which have a crystal direction being close to a given specimen direction (or sveral specimen directions).

```
mtexdata copper

% define which crystal direction to consider
h = Miller(1,1,1,ebsd('c').CS)

%  reference direction e.g. (1,2,3) transformed in crystal coordinates
refc = inv(ebsd('c').orientations) * vector3d(1,0,1);

% angle between refc and h be within e.g. 10*degree
cond1 = angle(refc,h) <= 10*degree;

plot(ebsd,ebsd.orientations)
nextAxis
plot(ebsd(cond1),ebsd(cond1).orientations)

% you can OR several consitions for e.g. a different reference direction
refc2 = inv(ebsd('c').orientations) * vector3d(1,0,0);
cond2 = angle(refc2,h) <= 10*degree;

nextAxis
plot(ebsd(cond1 | cond2),ebsd(cond1 | cond2).orientations)
```

Cheers,
Rüdiger

**Comment by kopparthiravikiran** [comment]
Hi. I think this is what I need. It worked with the copper file you gave as an example. However, I am getting an error when I am doing it on my file. My code is below.



%% Import Script for EBSD Data
%
% This script was automatically created by the import wizard. You should
% run the whoole script or parts of it in order to import your data. There
% is no problem in making any changes to this script.

%% Specify Crystal and Specimen Symmetries

% crystal symmetry
CS = {... 
  'notIndexed',...
  crystalSymmetry('m-3m', [2.9 2.9 2.9], 'mineral', 'Iron-alpha'),...
  'notIndexed',...
  'notIndexed',...
  'notIndexed'};

% plotting convention
setMTEXpref('xAxisDirection','east');
setMTEXpref('zAxisDirection','intoPlane');

%% Specify File Names

% path to files
pname = 'C:\Users\New Owner\Desktop\EBSD';

% which files to be imported
fname = [pname '\1.cpr'];

%% Import the Data

% create an EBSD variable containing the data
ebsd = EBSD.load(fname,CS,'interface','crc');

%%

% define which crystal direction to consider
h = Miller(1,1,1,ebsd('indexed').CS)

%  reference direction e.g. (1,2,3) transformed in crystal coordinates
refc = inv(ebsd('indexed').orientations) * vector3d(1,0,0);

% angle between refc and h be within e.g. 10*degree
cond1 = angle(refc,h) <= 10*degree;

plot(ebsd('indexed'),ebsd('indexed').orientations)
nextAxis
plot(ebsd(cond1),ebsd(cond1).orientations)

**Reply by kilir** [reply]
Try `ebsd = ebsd('Iron-alpha')`
Cheers,
Rüdiger
